preload/link-header-preload-non-html.html

```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>Makes sure that Link headers preload resources</title>
+<!--
+  This and the line below ensure that the trailing crossorigin in the link
+  header is honored, otherwise we'd load this resource twice and the test would
+  fail.
+-->
+<meta name="timeout" content="long">
+<link rel="preload" as="style" crossorigin href="resources/dummy.css?link-header-crossorigin-preload2">
+<link rel="stylesheet" crossorigin href="resources/dummy.css?link-header-crossorigin-preload2">
+<script src="resources/dummy.js?link-header-preload2"></script>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/preload/resources/preload_helper.js"></script>
+<body>
+<script>
+    const preloadLink = '/preload/resources/dummy.css';
+    function url_with_preload(content, type) {
+        const link = encodeURIComponent(`<${preloadLink}>;rel=preload;as=style`)
+        return getAbsoluteURL(`./resources/echo-preload-header.py?type=${type}&content=${encodeURIComponent(content)}&link=${link}`);
+    }
+
+    function test_document_type(content, type) {
+        async_test(t => {
+            const url = url_with_preload(content, type);
+            const win = window.open(url);
+            t.add_cleanup(() => win.close());
+            win.addEventListener('load', () => {
+                t.step_timeout(() => {
+                    const link = win.document.createElement('link');
+                    link.rel = 'preload';
+                    link.as = 'style'
+                    link.href = preloadLink;
+                    link.addEventListener('load', t.step_func(() => {
+                        const resources = win.performance.getEntriesByName(getAbsoluteURL(preloadLink));
+                        assert_equals(resources.length, 1);
+                        t.done();
+                    }));
+                    document.head.appendChild(link);
+                }, 500);
```

Comment:
Why "500"? Maybe you can use a PerformanceObserver to know when the preload link is actually done instead of a timeout?

Reply:
Revised.